.github/workflows/coverity.yml

```diff
@@ -0,0 +1,89 @@
+name: Coverity
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * 0'
+
+permissions: read-all
+
+jobs:
+  coverity:
+    name: Coverity
+    runs-on: [Linux, build]
+    container:
+      image: ghcr.io/intel/llvm/ubuntu2404_intel_drivers:alldeps
+      options: -u 1001:1001
+    env:
+      CCACHE_DIR: ${{ inputs.build_cache_root }}/build_cache_${{ inputs.build_cache_suffix }}
+      CCACHE_MAXSIZE: 8G
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        sparse-checkout: |
+          devops/actions
+
+    - name: Register cleanup after job is finished
+      uses: ./devops/actions/cleanup
+
+    - uses: ./devops/actions/cached_checkout
+      with:
+        path: src
+        ref: ${{ github.sha }}
+        cache_path: "/__w/repo_cache/"
+
+    - name: Get coverity tool
+      run: |
+        wget https://scan.coverity.com/download/linux64 --post-data "token=${{ secrets.COVERITY_TOKEN }}&project=intel%2Fllvm" -O coverity_tool.tgz
+        tar -xf coverity_tool.tgz
+
+    - name: Configure
+      env:
+        CC: gcc
+        CXX: g++
+        CUDA_LIB_PATH: "/usr/local/cuda/lib64/stubs"
+      run: |
+        mkdir -p $CCACHE_DIR
```

Comment:
I'm not familiar with ccache and how we use it in our CI. But I have some doubts regarding how the compiler should be built in this case. I assume for coverity we need a full build from scratch at least for the first time. May be the next builds can be incremental, but only if the previous build is used. It there a guarantee that the previous build will be used? Or should we just build it without ccache?

Reply:
https://documentation.blackduck.com/bundle/coverity-docs/page/coverity-analysis/topics/using_ccache_or_distcc.html

Reply:
I decided to drop ccache, I guess we can afford it for weekly runs on Sunday mornings. ccache makes it more complicated and I'm not sure what will be the result if there is a different machine from run to run or the cache is cleaned up.